### PR TITLE
LPS-118542

### PR DESCRIPTION
**Pull request by yuhai**
@dantewang, I use FileUtils.writeStringToFile() to clear log files. In wins, file.delete() can't delete opened file. In addition, I keep the origin commit and hope you can check clearly.

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

File: modules/apps/portal/portal-log4j-output-message-test/src/testIntegration/java/com/liferay/portal/log4j/output/message/test/Log4JOutputMessageTest.java [hunk 1]
```diff
@@ -36,6 +36,7 @@
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
+import org.apache.commons.io.FileUtils;
```
Comment (dantewang) [inline modules/apps/portal/portal-log4j-output-message-test/src/testIntegration/java/com/liferay/portal/log4j/output/message/test/Log4JOutputMessageTest.java:39]: @yuhai Please avoid this. If the file can not be deleted on Windows then just clean up the content.

File: modules/apps/portal/portal-log4j-output-message-test/src/testIntegration/java/com/liferay/portal/log4j/output/message/test/Log4JOutputMessageTest.java [hunk 2]
```diff
@@ -77,52 +78,27 @@ public static void setUpClass() throws Exception {
...
+				String portalFileNamePattern =
+					portalTimeBasedRollingPolicy.getFileNamePattern();
 
-					xmlFileRollingFileAppender.activateOptions();
+				portalFileNamePattern = StringUtil.extractLast(
+					portalFileNamePattern, StringPool.SLASH);
 
-					logger.addAppender(xmlFileRollingFileAppender);
-				}
+				_setLoggerRollingFileAppender(
```
Comment (dantewang) [inline modules/apps/portal/portal-log4j-output-message-test/src/testIntegration/java/com/liferay/portal/log4j/output/message/test/Log4JOutputMessageTest.java:98]: @yuhai Can you centralize the file appender related logic into the private method? It's weird to keep part of it here and have another part in the private method.

File: modules/apps/portal/portal-log4j-output-message-test/src/testIntegration/java/com/liferay/portal/log4j/output/message/test/Log4JOutputMessageTest.java [hunk 3]
```diff
@@ -191,38 +143,60 @@ private static Logger _getLogger() {
...
 			String level, String message, Throwable throwable)
 		throws Exception {
 
-		for (File file : _tempLogDir.listFiles()) {
-			String fileName = file.getName();
+		Matcher matcher = _textFileNamePattern.matcher(_textLogFile.getName());
 
-			if (fileName.endsWith(".log")) {
-				Matcher matcher = _textFileNamePattern.matcher(fileName);
+		Assert.assertTrue(
```
Comment (dantewang) [inline modules/apps/portal/portal-log4j-output-message-test/src/testIntegration/java/com/liferay/portal/log4j/output/message/test/Log4JOutputMessageTest.java:181]: @yuhai With a second thought, it still looks meaningless to do the assertion on the files that we create.
Could you try this: remove the filename pattern assertion from this method, and put the filename pattern assertion in a separate test case (`@Test testXxxxx()` method)? Try to save portal's file appenders into fields and get the real/exact file path to assert.

This way we don't need to parse portal appender's pattern when creating the test appenders.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**